be/src/exprs/vectorized/json_functions.cpp

```diff
@@ -8,29 +8,45 @@
 #include <boost/algorithm/string.hpp>
 #include <boost/tokenizer.hpp>
 
+#include "column/column_builder.h"
 #include "column/column_helper.h"
 #include "column/column_viewer.h"
 #include "common/status.h"
+#include "exprs/vectorized/function_helper.h"
+#include "exprs/vectorized/jsonpath.h"
+#include "glog/logging.h"
+#include "gutil/strings/split.h"
 #include "gutil/strings/substitute.h"
+#include "udf/udf.h"
+#include "util/json.h"
+#include "velocypack/vpack.h"
 
 namespace starrocks::vectorized {
 
 // static const re2::RE2 JSON_PATTERN("^([a-zA-Z0-9_\\-\\:\\s#\\|\\.]*)(?:\\[([0-9]+)\\])?");
 // json path cannot contains: ", [, ]
-static const re2::RE2 JSON_PATTERN(R"(^([^\"\[\]]*)(?:\[([0-9]+|\*)\])?)");
+const re2::RE2 JSON_PATTERN(R"(^([^\"\[\]]*)(?:\[([0-9\:\*]+)\])?)");
```

Comment:
what the meaning for `$a.b[::]`?

Reply:
The modification is for `$.a.b[1:3]`, known as array slicing.

Reply:
It looks like this regular will match an unintended case. such as `$.a.b[::]`, `$.a.b[:*]`

Reply:
Well, this regex does not validate the unintended case. But `ArraySelector::parse` will do more validate work, these illegal jsonpath will get a `Invalid json path` error.

Reply:
As @ZiheLiu suggested, I have separated it into two regex to validate, as `SIMPLE_JSONPATH_PATTERN` and `JSONPATH_PATTERN`.